src/schema/userTickets/mutations.ts

```diff
@@ -344,22 +345,17 @@ builder.mutationField("claimUserTicket", (t) =>
               purchaseOrderByTickets,
               logger,
             });
-            const createdPurchaseOrder = await createInitialPurchaseOrder({
-              DB: trx,
-              userId: USER.id,
-              logger,
-            });
 
-            let claimedTickets: Array<typeof insertUserTicketsSchema._type> =
-              [];
+            const ticketTemplatesIds = Object.keys(purchaseOrderByTickets);
 
-            // TODO: Measure and consider parallelizing this.
-            for (const { quantity, ticketId } of Object.values(
-              purchaseOrderByTickets,
-            )) {
-              // We pull the ticket template to see if it exists.
-              const ticketTemplate = await trx.query.ticketsSchema.findFirst({
-                where: (t, { eq }) => eq(t.id, ticketId),
+            const [createdPurchaseOrder, ticketTemplates] = await Promise.all([
+              createInitialPurchaseOrder({
+                DB: trx,
+                userId: USER.id,
+                logger,
+              }),
+              trx.query.ticketsSchema.findMany({
```

Comment:
Se eliminó el bucle for que iteraba sobre cada ticket. Ahora procesamos los tickets en lote, lo que reduce el número de consultas a la base de datos y mejora el rendimiento.